accepted/2021/User-Experience/Apple-Silicon-UX.md

```diff
@@ -0,0 +1,254 @@
+  - Modify the runtime probing to be consistent with the new `hostpolicy` probing.
+- Modify the build system to build the Universal Binaries
+- Modify installers
+  - Modify .NET 5 installers to allow installation without breaking universal `dotnet`.
+  - Modify .NET 6 installer to overwrite the x64 `dotnet`
+
+#### Open Questions
+
+- Would an alternative design be better? For instance:
+  - Choose a simpler design
+  - Use same basic universal `dotnet` and `hostfxr` ideas
+  - Keep separate runtimes. Each can optionally be installed without overwriting the other
+  - Move macOS .NET 6 CoreCLR runtime into a RID specific directory
+  - Teach `hostfxr` about the new `RID` in the probing for `hostpolicy`
+    - If the current architecture is not present treat it as a missing runtime.
+  - Probably more consistent with the original scope of our .NET 6
+  Apple Silicon plan.
+
+- What unanticipated consequences will a universal runtime have?
+
+- How will a universal runtime affect macOS publishing?
+
+#### Dotnet Muxer Side-By-Side Design Alternatives
+
+##### Separate install directories
+
+For other OS architectures CoreCLR has dealt with this by using separate install directories.  The default architecture would be on the path and the IDEs would know how to select the architecture as needed.
+
+##### Universal binaries
+
+There are a some design choices here:
+
+- How much runtime is universal?
+  - Dotnet muxer & hosfxr (library responsible for runtime selection)
+  - All runtime native binaries (starting at .NET 6)
+  - All runtime native and _some_ managed binaries (starting at .NET 6)
+  - All runtime native and _all_ managed binaries (starting at .NET 6)
+- How does `hostfxr` we identify which runtimes support which architecture?
+  - Let the kernel fatally fail
+  - Fail and retry with another arch (slow, not scalable)
+  - Policy heuristic
+    - .NET 5 runtime = x64
+    - .NET 6 runtime must be native
+    - .NET 6 runtime has RID in some well defined place
+    - .NET 6 runtime must be universal
+  - New config variable in runtime's json config file
+  - Runtime install path. Add RID to runtime install path.
+  - Universal `hostpolicy` with RID aware probing paths
+
+### Apple Silicon requires Signed Binaries
+
+Given that the SDK needs to handle this automatically for some platforms, it seems best to handle this consistently and automatically on all platforms.
+
+Proposal is currently _Unplanned/Uncommitted._
+
+### Publishing Universal Binary Apps
+
+No planned support for publishing CoreCLR universal binaries in .NET 6.
+
+This represent the current CoreCLR runtime team plan.
+
+Pursuing a universal CoreCLR runtime would affect this design and might allow us to at least specify a manual method to produce a universal app.
```

Comment:
I'd be better to fold this into https://github.com/dotnet/designs/pull/182/files#diff-f9955dc1a401114b9383fdfecab5a55da35312f33af1af21d4d46a0246fa7e16R158 section. It also needs clarification of what is exactly not supported.